src/com/goide/inspections/GoAssignmentToReceiverInspection.java

```diff
@@ -34,19 +31,45 @@
   protected GoVisitor buildGoVisitor(@NotNull ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
     return new GoVisitor() {
       @Override
-      public void visitReferenceExpression(@NotNull GoReferenceExpression o) {
-        super.visitReferenceExpression(o);
-        if (o.getReadWriteAccess() == ReadWriteAccessDetector.Access.Write) {
-          PsiElement resolve = o.resolve();
-          if (resolve instanceof GoReceiver) {
-            String message = "Assignment to method receiver doesn't propagate to other calls";
-            if (((GoReceiver)resolve).getType() instanceof GoPointerType) {
-              message = "Assignment to method receiver propagates only to callees but not to callers";
+      public void visitAssignmentStatement(@NotNull GoAssignmentStatement o) {
+        super.visitAssignmentStatement(o);
+        for (GoExpression expr : o.getLeftHandExprList().getExpressionList()) {
+          if (expr instanceof GoReferenceExpression) {
+            GoReferenceExpression refExpr = (GoReferenceExpression)expr;
+            PsiElement resolve = refExpr.resolve();
+            if (resolve instanceof GoReceiver) {
+              if (!canInspect(refExpr, (GoReceiver)resolve)) return;
+              String message = "Assignment to method receiver doesn't propagate to other calls";
+              if (((GoReceiver)resolve).getType() instanceof GoPointerType) {
+                message = "Assignment to method receiver propagates only to callees but not to callers";
+              }
+              holder.registerProblem(expr, message, WEAK_WARNING);
             }
-            holder.registerProblem(o, message, WEAK_WARNING);
           }
         }
       }
     };
   }
-}
\ No newline at end of file
+
+  private static boolean canInspect(GoReferenceExpression o, @Nullable GoReceiver receiver) {
+    if (receiver == null) return false;
+    GoType resolveType = receiver.getType();
+    if (resolveType == null) return false;
+    PsiElement parent = o.getParent();
+
+    if (resolveType instanceof GoPointerType) {
+      if (parent instanceof GoUnaryExpr) {
+        if (((GoUnaryExpr)parent).getMul() != null) return false;
+      }
+    }
+
+    GoType underlyingType = resolveType.getUnderlyingType();
+    if (underlyingType instanceof GoChannelType ||
```

Comment:
I've replaced it with `return !(() || () || ())` but not sure if it's preferred to `return (!() && !() && !())`